include/aws/auth/credentials.h

```diff
@@ -290,6 +276,35 @@ struct aws_credentials_provider_ecs_options {
     uint32_t port;
 };
 
+/*
+ * Configuration options for the provider that sources credentials from ECS container metadata.
+ * The ECS creds provider can be used to access creds via either a relative URI to a fixed endpoint
+ * (http://169.254.170.2) or via a full URI specified by environment variables (in order of priority):
+ * AWS_CONTAINER_CREDENTIALS_RELATIVE_URI
+ * AWS_CONTAINER_CREDENTIALS_FULL_URI
+ *
+ * For the Authorization token, there are two ways (in order of priority):
+ * 1. AWS_CONTAINER_AUTHORIZATION_TOKEN_FILE (an env var which contains the absolute path to the token file. The file
+ * will be re-read for each call to get credentials.)
+ * 2. AWS_CONTAINER_AUTHORIZATION_TOKEN (an env var that contains a static auth token)
+ */
+struct aws_credentials_provider_ecs_environment_options {
```

Comment:
trivial/debatable: declare the environment-options struct before the explicit-options struct, so that people encounter it first when searching, since it's the more common use case

likewise, declare the new_ecs_from_environment() function before the new_ecs() function